### PR TITLE
[SofaCUDA] Fix some namespace, headers includes and decl exports

### PR DESCRIPTION
**Pull request by epernod**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaLineModel.cpp [hunk 1]
```diff
@@ -19,8 +19,11 @@
 *                                                                             *
 * Contact information: contact@sofa-framework.org                             *
 ******************************************************************************/
+#ifndef SOFA_GPU_CUDA_CUDALINEMODEL_CPP
+#define SOFA_GPU_CUDA_CUDALINEMODEL_CPP
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaLineModel.cpp:22]: you don't have to add a `ifndef` guard. Just define the variable

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaSphereModel.cpp [hunk 2]
```diff
@@ -19,8 +19,11 @@
 *                                                                             *
 * Contact information: contact@sofa-framework.org                             *
 ******************************************************************************/
+#ifndef SOFA_GPU_CUDA_CUDASPHEREMODEL_CPP
+#define SOFA_GPU_CUDA_CUDASPHEREMODEL_CPP
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaSphereModel.cpp:22]: Same here

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaTriangleModel.cpp [hunk 3]
```diff
@@ -19,8 +19,11 @@
 *                                                                             *
 * Contact information: contact@sofa-framework.org                             *
 ******************************************************************************/
+#ifndef SOFA_GPU_CUDA_CUDATRIANGLEMODEL_CPP
+#define SOFA_GPU_CUDA_CUDATRIANGLEMODEL_CPP
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaTriangleModel.cpp:22]: Same here

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaTriangleModel.h [hunk 4]
```diff
@@ -22,18 +22,30 @@
 #pragma once
 
 #include <sofa/gpu/cuda/CudaTypes.h>
-#include <SofaMeshCollision/TriangleModel.h>
-#include <SofaMeshCollision/TriangleModel.inl>
+#include <sofa/component/collision/geometry/TriangleModel.h>
```
Comment (alxbilger) [inline applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaTriangleModel.h:25]: The inl is not required here?

**Review by alxbilger — CHANGES_REQUESTED** [review]

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaTriangleModel.h (hunk 4 above)
Comment (epernod) [reply applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaTriangleModel.h:25]: not sure, we only add alias in this header. The inl is already included in this class .cpp

Comment (damienmarchal) [reply applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaTriangleModel.h:25]: I have the impression that, in general for template specialization, there is the following inclusions:
- specialized_template.h includes base_template.h, 
- specialized_template.inl includes base_template.inl, 
- specialized_template.cpp includes the specialized_template.inl

**Comment by damienmarchal** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Pull request merged**